Source/Basetypes/BigFloat.cs

```diff
@@ -157,6 +157,56 @@ public static BigFloat FromBigDec(BigDec v, int significandSize, int exponentSiz
       return new BigFloat(isNeg, sig, exp, sigSize, expSize);
     }
 
+    [Pure]
+    public static BigFloat FromBVString(String s)
+    {
+      /*
+       * String must be either of the format (fp (bv* *) (bv* *) (bv* *))
+       * or of the special value format: (_ val * *)
+       * Where * indicates an integer value
+       * and val is one of the special values: NaN, +oo, -oo, zero, -zero
+       */
+      BIM sig, exp;
+      int sigSize, expSize;
+      bool isNeg;
+
+      String[] args;
+
+      if (s.IndexOf("_") == 1)
+      {
+        s = s.Substring(0, s.Length - 1);
+        args = s.Split(' ');
+        sigSize = int.Parse(args[3]);
+        expSize = int.Parse(args[2]);
+        if (sigSize <= 0 || expSize <= 0)
+          throw new FormatException("Significand and Exponent sizes must be greater than 0");
+        return new BigFloat(args[1], sigSize, expSize);
+      }
+
+      args = s.Split('_');
```

Comment:
Hmmm, I am confused here. In the comment above you say that format is `(fp (bv* *) (bv* *) (bv* *))`, but then here you are splitting on `_`, which is not even in this format. What am I missing here?

Reply:
The description was wrong, fixed.